### PR TITLE
add available images to configmap

### PR DESCRIPTION
**Pull request by jinchihe**
**What this PR does / why we need it**:
Today I investigated the issue #296 , seems that works for last docker image, but the user cannot input the last docker image as `RuntimeVersion must be one of latest`, and `latest` image is not last one.

```
(python36) [root@jinchi1 kfserving]# kubectl create -f /opt/kubeflow/shared/kfserving/docs/samples/pytorch/pytorch.yaml
Error from server: error when creating "/opt/kubeflow/shared/kfserving/docs/samples/pytorch/pytorch.yaml": admission webhook "kfservice.kfserving-webhook-server.validator" denied the request: RuntimeVersion must be one of latest
```
We should add more available images so that user can configure the docker image tag while define the kfservice. 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kfserving/319)

### CODE REVIEW

File: pkg/apis/serving/v1alpha2/framework_scikit.go [hunk 1]
```diff
@@ -26,6 +26,9 @@ import (
 var (
 	AllowedSKLearnRuntimeVersions = []string{
 		"latest",
+		"v0.1.0",
+		"v0.1.1",
+		"v0.1.2",
```
[the remark this answers is not in the document]
Comment (jinchihe) [reply pkg/apis/serving/v1alpha2/framework_scikit.go]: I got from here: https://console.cloud.google.com/gcr/images/kfserving/GLOBAL/pytorchserver?project=kfserving&organizationId=433637338589&gcrImageListsize=30

**Comment by jinchihe** [comment]
> how are we dealing this with images in other parts of kubeflow umbrella

Checked some parts, they are not using the `Google Cloud Builder` to automatically build the image , the kfservice iamges are built by `Google Cloud Builder` and triggerred by github tag.

**Comment by kkasravi** [comment]
@animeshsingh @ellis-bigelow @jinchihe 
I've been working on e2e testing and have implemented a tektoncd pipeline to do the ci/cd (build-push-deploy-test) for kfctl. The build uses kaniko. I'm also replacing ksonnet with kustomize for the argo Workflows. 

Here's a good [article](https://medium.com/@ndeloof/introduction-to-tektoncd-57114b8ff52b) on tektoncd and an early [draft](https://docs.google.com/document/d/1oaBBJerOkKIuAAn_Swbu8uVBPtaYgvAVbQV1NxTsSSg/edit#heading=h.9g4gb5dvlquq) of the e2e design

**Comment by ellis-bigelow** [comment]
@kkasravi this is amazing!! 
+1 for Tekton, Kaniko, and Kustomize

Really looking forward to integrating here.

**Comment by kkasravi** [comment]
thanks @ellis-bigelow !

**Comment by yuzisun** [comment]
@kkasravi I also have been evaluating tekton recently, let's have a chat!

**Comment by animeshsingh** [comment]
@yuzisun who will be updating config map? and what's the point of spending more effort temporarily when its going to be replaced?

**Comment by yuzisun** [comment]
I am not sure if the build system solves this problem though, like for example if we change the data plane in 0.2 release all previous built pytorch image won’t be compatible with control plane, so the list of supported image versions need to be carefully put into the config map.

**Comment by jinchihe** [comment]
@yuzisun Yes, it perhaps be a good idea to put the tag in the configmap. I'm wonder how to update the configmap before cutting new branch/tags? Any good practice to add to configmap automatically while cutting or need to manully update that before cutting? Thanks.

**Comment by animeshsingh** [comment]
well, i still don't see a strong reason why these need to go in configmap ?and what's the harm in putting in plain text? 

Also In general, can we just pin to the latest working image we support? Even in Kubeflow pipelines throughout, the latest image is automatically build as part of the release, tagged, and mentioned in plain text
https://github.com/kubeflow/pipelines/blob/master/components/kubeflow/katib-launcher/component.yaml

**Comment by rakelkar** [comment]
I think this is to do with framework versions that we support. It is linked to build for the inference servers that we happen to build. But for those that we dont build, the supported version list is a function of testing.

I think it makes a lot of sense to put it into a config map so that users are not blocked on a code change if they want to make a particular version available in their cluster.

**Comment by yuzisun** [comment]
For the model servers we build we may just need to support the current release version, not sure if it is worth to allow user to run previous versions by default which might run into backwards compatible issues for data plane changes. I suggest we keep the runtime versions in config map and update the image version every time we do a new release, if user really want to go for previous versions they can still modify config map to add the versions they want to the list.

**Comment by jinchihe** [comment]
@yuzisun Thanks, will address this week once get chance.

**Comment by jinchihe** [comment]
@yuzisun as we discussed offline, I logged the ticket #334 to trace add the runtime image to configmap, and will handle that later once get chance. For now, let's just keep the updates last runtime in the go file so that user can use new tag image. I will update that once get chance (and I'm going to work on the new CI/CD testing framework, may be have new changes about this). 

For each release we should update that before cutting release branch or tag.  Thanks.

Apologize for the late update as I'm running for other urgent things recent days.

**Comment by yuzisun** [comment]
thanks @jinchihe !
/unhold
/lgtm

**Pull request merged**